### PR TITLE
Uploads basic

### PR DESCRIPTION
**Pull request by moshiko1988**
Implements upload resource with user ownership.

solves #1

### CODE REVIEW

File: app/models/upload.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const mongoose = require('mongoose');
+
+const uploadSchema = new mongoose.Schema({
+  tag: [{
+    type: String,
```
Comment (ddobson) [inline app/models/upload.js]: Shouldn't `tag` be an array?

File: app/models/upload.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    lowercase: true,
```
Comment (ddobson) [inline app/models/upload.js]: Does `lowercase: true` normalize input to lowercase or does it just reject uppercase characters?

File: app/models/upload.js (hunk 1 above)
Comment (Sofistication) [inline app/models/upload.js]: It is probably not clear due to the weirdness of the mongoose syntax we needed to use in order to force `tag` to be lowercase, but it is an array.

The square brackets around the object are what define it as an array, the object allows us to enforce certain validations on the inner elements of the array.

File: app/models/upload.js (hunk 2 above)
Comment (Sofistication) [inline app/models/upload.js]: We tested this with the script in `scripts/models` and it normalizes, not rejects.

**Pull request merged**